services/arrow-flight/src/main/java/com/dremio/service/flight/DremioFlightSqlProducer.java

```diff
@@ -261,14 +261,25 @@ public void getStreamCatalogs(CallContext callContext, Ticket ticket,
   public FlightInfo getFlightInfoSchemas(CommandGetSchemas commandGetSchemas,
                                          CallContext callContext,
                                          FlightDescriptor flightDescriptor) {
-    throw CallStatus.UNIMPLEMENTED.withDescription("CommandGetSchemas not supported.").toRuntimeException();
+    final Schema schema = getSchemaSchemas().getSchema();
```

Comment:
This looks similar to the catalogs implementation, can you refactor to reduce duplication?

Reply:
Fixed